src/nix/run.cc

```diff
@@ -135,6 +135,7 @@ struct CmdShell : InstallablesCommand, MixEnvironment
         unixPath.insert(unixPath.begin(), pathAdditions.begin(), pathAdditions.end());
         auto unixPathString = concatStringsSep(":", unixPath);
         setenv("PATH", unixPathString.c_str(), 1);
+        setenv("IN_NIX_SHELL", "impure", 1);
```

Comment:
If going the envvar route it would probably be nicer to keep the `NIX_` prefix for consistency with other varnames: https://nixos.wiki/wiki/Environment_variables

Reply:
Oh I see from the context link this is a well established variable (which I didn't know about!). Never mind 🤷  I guess that horse has left the barn.